File: 0055-jump-game/0055-jump-game.cs
// Note: i looked at the neetcode.io video first 
// translating my python->c++->java->c->javascript code to C# 
// unlike the others i've never actually taken a course in C# 
// or used it other than for a few leetcode problems 
public class Solution {
    public bool CanJump(int[] nums) {
        int goal = nums.Length - 1; 
        for (int i = goal; i >= 0; i--) {
            if (i + nums[i] >= goal) {
                goal = i;
            }
        }
        return (goal == 0);         
    }
}

File: 0055-jump-game/0055-jump-game.java
// Note I looked at the neetcode solution first,
// translating my python->c++ code to java 
class Solution {
    public boolean canJump(int[] nums) {
        int goal = nums.length - 1; 
        for (int i = goal; i >= 0; i--) {
            if (i + nums[i] >= goal) {
                goal = i;
            }
        }
        return (goal == 0); 
        
    }
}
